### ISSUE TITLE
pkgconfig files do not define providers directory

### ISSUE THREAD

**Issue by simo5** [openssl/openssl#18516]
pkgconfig files like libcrypto.pc are used by other software to determine where certain files are or need to be installed
For example libcrypto.pc reports the location where engines should be installed
```
$ pkgconf libcrypto --print-variables
enginesdir
includedir
libdir
exec_prefix
prefix
pcfiledir
```
So the engines location can be queried:
```
$ pkgconf libcrypto --variable=enginesdir
/usr/lib64/engines-3
```

However there is no variable to know where to install providers.

openssl -a does report that value
```
openssl version -a |grep MODULESDIR
MODULESDIR: "/usr/lib64/ossl-modules"
```

It should be made available via pkgconfig files too

**Comment by beldmit** [comment]
I think lack of this information in libcrypto.pc is similar to a missing accessor so the fix, when happens, should go to 3.0

**Comment by levitte** [comment]
This was fixed in master by #17605, so a backport to 3.0 is a really simple matter of a cherry-pick

**Comment by levitte** [comment]
3.0 only fix in #18520

**Comment by simo5** [comment]
Great!
Thank you

**Comment by nhorman** [comment]
marking as inactive, to be closed at the end of 3.4 dev barring further input
